nomad/job_endpoint_validators_test.go

```diff
@@ -0,0 +1,15 @@
+package nomad
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestJobNamespaceConstraintCheckHook_Name(t *testing.T) {
+	t.Parallel()
+
+	require.Equal(t, "namespace-constraint-check", new(jobNamespaceConstraintCheckHook).Name())
+}
+
+// TODO: More tests
```

Comment:
The code in `nomad/job_endpoint_validators.go` looks like it would be nicely served here by a "table driven" test where you pass a list of task/namespace pairs and whether they should validate or not.